gnomad_qc/v4/annotations/generate_freq.py

```diff
@@ -290,92 +319,96 @@ def generate_freq_and_hists_ht(
+        ds_pop_counts=hl.eval(ds_ht.ds_pop_counts),
+    )
+
+    logger.info("Densifying VDS...")
     mt = hl.vds.to_dense_mt(vds)
 
     logger.info("Computing sex adjusted genotypes...")
-    mt = mt.transmute_entries(
-        GT=adjusted_sex_ploidy_expr(mt.locus, mt.GT, mt.sex_karyotype),
+    ab_expr = mt.AD[1] / mt.DP
+    gt_expr = adjusted_sex_ploidy_expr(mt.locus, mt.GT, mt.sex_karyotype)
+    mt = mt.select_entries(
+        "DP",
+        "GQ",
+        GT=gt_expr,
+        adj=get_adj_expr(gt_expr, mt.GQ, mt.DP, mt.AD),
+        _het_ab=ab_expr,
+        _high_ab_het_ref=(ab_expr > ab_cutoff) & ~mt._het_non_ref,
     )
-    mt = mt.annotate_entries(adj=get_adj_expr(mt.GT, mt.GQ, mt.DP, mt.AD))
 
-    logger.info("Annotating frequencies and counting high AB het calls...")
-    additional_strata_expr = [
-        {"gatk_version": mt.gatk_version},
-        {"gatk_version": mt.gatk_version, "pop": mt.pop},
-        # TODO: confirm this is how we want to name this.
-        {"ukb_sample": mt.ukb_sample},
-    ]
+    def _high_ab_het(entry, col):
+        """
+        Add description.
 
-    def _needs_high_ab_het_fix(entry, col):
+        :param entry:
+        :param col:
+        :return:
+        """
         return hl.int(
             entry.GT.is_het_ref()
-            # & (entry._het_ad / entry.DP > ab_cutoff)
-            & (entry.AD[1] / entry.DP > ab_cutoff)
             & entry.adj
             & ~col.fixed_homalt_model
-            & ~entry._het_non_ref
-        )  # Skip adjusting genotypes if sample originally had a het nonref genotype.
-
-    freq_ht = annotate_freq(
-        mt,
-        sex_expr=mt.sex_karyotype,
-        pop_expr=mt.pop,
-        downsamplings=hl.eval(mt.downsamplings),
-        downsampling_expr=mt.downsampling,
-        ds_pop_counts=hl.eval(mt.ds_pop_counts),
-        additional_strata_expr=additional_strata_expr,
-        entry_agg_funcs={"high_ab_hets_by_group": (_needs_high_ab_het_fix, hl.agg.sum)},
-        annotate_mt=False,
-    )
-
-    logger.info("Making freq index dict...")
-    # Add additional strata to the sort order, keeping group, i.e. adj, at the end.
-    sort_order = deepcopy(SORT_ORDER)
-    sort_order[-1:-1] = ["gatk_version", "ukb_sample"]
-
-    freq_ht = freq_ht.annotate_globals(
-        freq_index_dict=make_freq_index_dict_from_meta(
-            freq_meta=freq_ht.freq_meta,
-            label_delimiter="_",
-            sort_order=sort_order,
+            & entry._high_ab_het_ref
         )
+
+    logger.info("Annotating frequencies and counting high AB het calls...")
+    freq_ht = compute_freq_by_strata(
+        mt.annotate_cols(group_membership=ht[mt.col_key].group_membership),
+        entry_agg_funcs={"high_ab_hets_by_group": (_high_ab_het, hl.agg.sum)},
     )
+    freq_ht = freq_ht.annotate_globals(**ht.index_globals())
+
     logger.info("Setting Y metrics to NA for XX groups...")
+    freq_ht = annotate_freq_index_dict(freq_ht)
```

Comment:
The `annotate_freq_index_dict`? I replaced the call to `make_freq_index_dict_from_meta` from above with this

Reply:
Sorry, I wasnt clear. I was thinking the output of this isnt used because we recall the function in the combine step but I guess we have it so the split/combine is optional so for the code to be complete we should build this here too.